### PR TITLE
TZ022 - [OMS] End to end flow for adding a product to a cart

### PR DESCRIPTION
**Pull request by Andrei0205**
- Create Cart
- Add item to cart
- Handle validation exceptions

### CODE REVIEW

File: tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package com.nagarro.tazzan.ordermanagement.controller;
+
+public class OrderController {
```
Comment (cornelBilica) [inline tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/controller/OrderController.java:3]: why do you need this now if it's empty?

File: tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/exceptions/GlobalResponseException.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                                                                  HttpStatus status, WebRequest request) {
+        BindingResult bindingResult = ex.getBindingResult();
+        List<ErrorResponse> errors = bindingResult.getFieldErrors()
+                .stream()
+                .map(fieldError -> new ErrorResponse(fieldError.getField(), fieldError.getDefaultMessage()))
+                .collect(Collectors.toList());
+        return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+    }
+
+    //@Override
```
Comment (cornelBilica) [inline tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/exceptions/GlobalResponseException.java:31]: remove the commented code

File: tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/configuration/BeanConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import org.modelmapper.convention.MatchingStrategies;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BeanConfiguration {
+    @Bean
+    public ModelMapper getModelMapper() {
+        ModelMapper modelMapper = new ModelMapper();
+        modelMapper.getConfiguration().setMatchingStrategy(MatchingStrategies.STRICT);
```
Comment (alxdbvwork) [inline tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/configuration/BeanConfiguration.java:13]: why do you need this?

File: tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/exceptions/CartAlreadyExistException.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.nagarro.tazzan.ordermanagement.exceptions;
+
+public class CartAlreadyExistException extends RuntimeException{
```
Comment (alxdbvwork) [inline tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/exceptions/CartAlreadyExistException.java:3]: rename to DuplicateCartException

File: tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/facade/CartConverter.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+@Component
+@RequiredArgsConstructor
+public class CartConverter {
+
+    private final ModelMapper modelMapper;
+
+    public Cart convertCartDtoToEntity(CartDto cartDto) {
+        return modelMapper.map(cartDto, Cart.class);
+    }
+
```
Comment (alxdbvwork) [inline tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/facade/CartConverter.java:18]: extra space?

File: tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/facade/CartFacade.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CartFacade {
+    private final CartService cartService;
+    private final CartConverter cartConverter;
+    private final CartItemConverter cartItemConverter;
+
```
Comment (alxdbvwork) [inline tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/facade/CartFacade.java:17]: same here

File: tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/persistence/repository/CartRepositoryGateway.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class CartRepositoryGateway {
+
+    private final CartRepository cartRepository;
+
+    public Cart createCart(Cart cart) {
+        return cartRepository.save(cart);
```
Comment (alxdbvwork) [inline tazzan/orderManagement/src/main/java/com/nagarro/tazzan/ordermanagement/persistence/repository/CartRepositoryGateway.java:16]: pazea ca vine Cristi cu tacu sa te bata

**Review by cristianiova — APPROVED** [review]

**Pull request merged**